### PR TITLE
DEV-37: Lint custom javascript with eslint

### PR DESCRIPTION
**Pull request by jesconstantine**
This PR:

* adds a new phing target for installing npm dependencies for the project and for Drupal core
* adds this new phing target as a dependency for the existing `code-review` target
* adds eslint as a step to the `code-review` target
* removes `.js` files from `phpmd.xml` to ensure PHP linters are linting js files

Test this PR with https://github.com/palantirnet/drupal-skeleton/pull/146

### CODE REVIEW

File: defaults/install/build.xml [hunk 1]
```diff
@@ -127,6 +127,10 @@
 
         <!-- Run Drupal Check. -->
         <foreach list="${drupal-check.directories}" param="drupal-check.dir" target="drupal-check" />
+
+        <!-- Run eslint on custom javascript -->
+        <echo msg="$> npx eslint --ext .js ${drupal.root}/modules/custom" />
+        <exec command="npx eslint --ext .js --no-error-on-unmatched-pattern ${drupal.root}/modules/custom" logoutput="true" checkreturn="true" />
```
Comment (jesconstantine) [inline defaults/install/build.xml]: Should we also be linting custom theme files?

Comment (agentrickard) [reply defaults/install/build.xml]: It depends -- those are generally in a styleguide which should handle its own linting. 

Perhaps a config option in `build.yml` to include theme linting?

Comment (agentrickard) [reply defaults/install/build.xml]: A config is what I did in DHS -- https://github.com/palantirnet/wisdhs-workweb/pull/411/files

File: targets/install.xml [hunk 2]
```diff
@@ -280,7 +280,14 @@
...
                 <!-- @see https://docs.acquia.com/cloud-platform/performance/memcached/enable/ -->
-                <httpget url="https://docs.acquia.com/_downloads/cloud-memcache-d8+.php" dir="${build.dir}/${drupal.root}/sites/${drupal.site.dir}" filename="settings.acquia-memcache.php"/>
+                <composer command="require" composer="${composer.composer}">
+                    <arg line="--working-dir ${application.startdir}"/>
+                    <arg value="drupal/memcache"/>
+                </composer>
+                <composer command="require" composer="${composer.composer}">
+                    <arg line="--working-dir ${application.startdir}"/>
+                    <arg value="acquia/memcache-settings"/>
+                </composer>
```
Comment (jesconstantine) [inline targets/install.xml]: We can roll this back once https://github.com/palantirnet/the-build/pull/226 is merged but this unblocks the build installer test step.

**Review by patrickfweston — APPROVED** [review]
LGTM and the change in the `drupal-skeleton` project looks great!

**Pull request merged**